homeassistant/components/alarm_control_panel/totalconnect.py

```diff
@@ -0,0 +1,86 @@
+"""Interfaces with TotalConnect alarm control panels."""
+import logging
+import voluptuous as vol
+import homeassistant.components.alarm_control_panel as alarm
+from homeassistant.components.alarm_control_panel import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_PASSWORD, CONF_USERNAME, STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME, STATE_ALARM_DISARMED, STATE_UNKNOWN,
+    CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['total_connect_client==0.7']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'total-connect-client'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup a TotalConnect control panel."""
+    name = config.get(CONF_NAME)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    total_connect = TotalConnect(hass, name, username, password)
+    async_add_devices([total_connect])
```

Comment:
Rename this to `add_devices`, and pass `True`. This will call the `entity.update()` method before adding it to hass, so that the initial state is correct and we don't get extra state changes when hass is restarted.

```python
add_devices([total_connect], True)
```